src/Event/ReturnEvent.php

```diff
@@ -25,13 +27,31 @@ class ReturnEvent extends Event
     private $return;
 
     /**
+     * The request sent to the client
+     *
+     * @var RequestInterface
+     */
+    private $request;
+
+    /**
+     * The response from the client
+     *
+     * @var ResponseInterface
+     */
+    private $response;
+
+    /**
      * Constructor
      *
      * @param mixed $return
+     * @param RequestInterface|null $request
+     * @param ResponseInterface|null $response
      */
-    public function __construct($return)
+    public function __construct($return, RequestInterface $request = null, ResponseInterface $response = null)
```

Comment:
No, I'm sure this was done for BC reasons, but this event *shouldn't* be used outside the library.  In future versions that line will be a lot clearer.